API.md

```diff
@@ -640,7 +640,9 @@ const schema = Joi.any().strict();
 Sets a default value if the original value is undefined where:
 - `value` - the value.
   - `value` supports [references](#refkey-options).
-  - `value` may also be a function which returns the default value. If `value` is specified as a function that accepts a single parameter, that parameter will be a context object that can be used to derive the resulting value. **This clones the object however, which incurs some overhead so if you don't need access to the context define your method so that it does not accept any parameters**.
+  - `value` may also be a function which returns the default value. If `value` is specified as a function that accepts a single parameter, that parameter will be a context object that can be used to derive the resulting value.
+    - Use a function when setting a dynamic value, such as the current time. Ex: `default(() => new Date(), 'time of creation')`
```

Comment:
The example can just be `default(Date.now, 'time of creation')`. Adding another closure here doesn't do anything.

Reply:
That depends what you want, both are valid, but different types as output.